storage/innobase/xtrabackup/src/xbcloud/s3.cc

```diff
@@ -742,6 +742,19 @@ void S3_client::retry_error(Http_response *resp, bool *retry) {
     if (s3_resp.error_code() == "RequestTimeout") {
       *retry = true;
     }
+    if (s3_resp.error_code() == "ExpiredToken" ||
+        s3_resp.error_code() == "InvalidToken" ||
+        s3_resp.error_code() == "TokenRefreshRequired") {
+      if (api_version == S3_V4 && ec2_instance != nullptr &&
+          ec2_instance->get_is_ec2_instance_with_profile() &&
+          ec2_instance->fetch_metadata()) {
+        access_key = ec2_instance->get_access_key();
+        secret_key = ec2_instance->get_secret_key();
```

Comment:
Can access key and secret key change after expiry? may be it is OK to ask again. Just checking.

Reply:
They do not specify it. In case they do change, our code is prepared to handle.